core/types/transaction_signing.go

```diff
@@ -91,7 +40,16 @@ type sigCache struct {
 
 // MakeSigner returns a Signer based on the given chain config and block number.
 func MakeSigner(config *params.ChainConfig, blockNumber *big.Int) Signer {
-	return NewOVMSigner(config.ChainID)
+	var signer Signer
+	switch {
+	case config.IsEIP155(blockNumber):
+		signer = NewEIP155Signer(config.ChainID)
+	case config.IsHomestead(blockNumber):
+		signer = HomesteadSigner{}
+	default:
+		signer = FrontierSigner{}
```

Comment:
we never want to return anything besides `EIP155Signer`. This would result in a diff but would give extra assurance that we never use the wrong signer

Reply:
What impact would using the wrong signer have?

Reply:
Allowing transactions that are not protected with a chainid. All transactions must be protected with a chainid

Reply:
Transactions w/o chain ID should fail inside the `OVM_ECDSAContractAccount`.